Mathlib/Analysis/ODE/PicardLindelof.lean

```diff
@@ -10,397 +10,586 @@ import Mathlib.Topology.MetricSpace.Contracting
 /-!
 # Picard-Lindelöf (Cauchy-Lipschitz) Theorem
 
-In this file we prove that an ordinary differential equation $\dot x=v(t, x)$ such that $v$ is
-Lipschitz continuous in $x$ and continuous in $t$ has a local solution, see
-`IsPicardLindelof.exists_forall_hasDerivWithinAt_Icc_eq`.
-
-As a corollary, we prove that a time-independent locally continuously differentiable ODE has a
-local solution.
+We prove the (local) existence of integral curves and flows to time-dependent vector fields.
+
+Let `f : ℝ → E → E` be a time-dependent (local) vector field on a Banach space, and let `t₀ : ℝ`
+and `x₀ : E`. If `f` is Lipschitz continuous in `x` within a closed ball around `x₀` of radius
+`a ≥ 0` at every `t` and continuous in `t` at every `x`, then there exists a (local) solution
+`α : ℝ → E` to the initial value problem `α t₀ = x₀` and `deriv α t = f t (α t)` for all
+`t ∈ Icc tmin tmax`, where `L * max (tmax - t₀) (t₀ - tmin) ≤ a`.
+
+We actually prove a more general version of this theorem for the existence of local flows. If there
+is some `r ≥ 0` such that `L * max (tmax - t₀) (t₀ - tmin) ≤ a - r`, then for every
+`x ∈ closedBall x₀ r`, there exists a (local) solution `α x` with the initial condition `α t₀ = x`.
+In other words, there exists a local flow `α : E → ℝ → E` defined on `closedBall x₀ r` and
+`Icc tmin tmax`.
+
+The proof relies on demonstrating the existence of a solution `α` to the following integral
+equation:
+$$\alpha(t) = x_0 + \int_{t_0}^t f(\tau, \alpha(\tau))\,\mathrm{d}\tau.$$
+This is done via the contraction mapping theorem, applied to the space of Lipschitz continuous
+functions from a closed interval to a Banach space. The needed contraction map is constructed by
+repeated applications of the right hand side of this equation.
+
+## Main definitions and results
+
+* `picard f t₀ x₀ α t`: the Picard iteration, applied to the curve `α`.
```

Comment:
style nit: this is not a complete sentence, hence I'd not end this with a period